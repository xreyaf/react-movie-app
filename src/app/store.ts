import { configureStore } from "@reduxjs/toolkit";
import movieApi from "../features/movies/movieApi";

export const store = configureStore({
  reducer: {
    [movieApi.reducerPath]: movieApi.reducer
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware().concat(movieApi.middleware);
  }
});
export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
