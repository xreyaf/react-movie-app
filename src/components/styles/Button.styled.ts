import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const ButtonStyled = styled(motion.button)`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.5rem;
  outline: none;
  border: none;
  border-radius: 1rem;
  background: ${theme.colors.primary400};
  color: ${theme.colors.white100};
  padding: 1rem 2rem;
  font-size: 1rem;
  margin-bottom: .6rem;
  svg{
    align-self: baseline;
  }
`;
