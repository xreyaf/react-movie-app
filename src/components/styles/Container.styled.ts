import styled from "@emotion/styled";

export const Container = styled.div`
  width: 100%;
  max-width: 1440px;
  padding: 1rem 5rem;
  margin: 0 auto;
`;
