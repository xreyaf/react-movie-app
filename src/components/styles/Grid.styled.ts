import styled from "@emotion/styled";
import { motion } from "framer-motion";

export const GridStyled = styled(motion.div)`
  display: grid;
  grid-template-columns: repeat(4, minmax(0, 1fr));
  row-gap: 1.4rem;
  column-gap: 1.4rem;
  align-items: start;
  @media screen and (max-width: 1199.98px) {
    grid-template-columns: repeat(3, minmax(0, 1fr));
    transition: all .5s;
  }
  @media screen and (max-width: 991.98px) {
    row-gap: .8rem;
    column-gap: .8rem;
    transition: all .5s;
  }
  @media screen and (max-width: 816.98px) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
    row-gap: .8rem;
    column-gap: .8rem;
    transition: all .5s;
  }
  @media screen and (max-width: 575.98px) {
    grid-template-columns: repeat(1, minmax(0, 1fr));
    transition: all .5s;
  }
`;
