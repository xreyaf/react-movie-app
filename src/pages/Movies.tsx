import React from "react";
import { ContainerStyled } from "../components/styles/Container.styled";
import { GridStyled } from "../components/styles/Grid.styled";
import Card from "../components/Card";
import { useGetTrendingQuery } from "../features/movies/TMDBApi";

const Movies = () => {
  const { data } = useGetTrendingQuery('movie');
  return (
    <ContainerStyled>
      <h1>Movies page</h1>
      <br />
      <GridStyled>
        {data && <Card type='movie' props={data} />}
      </GridStyled>

    </ContainerStyled>
  );
};

export default Movies;
