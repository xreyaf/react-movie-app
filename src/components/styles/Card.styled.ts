import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const CardStyled = styled(motion.div)`
  position: relative;
  border-radius: 1rem;
  overflow: hidden;
  background-color: ${theme.colors.grey900_90};
  width: 100%;
  min-height: 100%;
`;
export const CardImageWrapper = styled.div`
  display: inline-block;
  border-radius: 1rem;
  width: 100%;
  height: 88%;
  overflow: hidden;
`;

export const ImageStyled = styled(motion.img)`
  object-fit: cover;
  width: 100%;
  min-height: 100%;
  vertical-align: middle;
`;

export const CardInfo = styled.div`
  padding: .5rem;
  margin-bottom: .5rem;
  overflow: hidden;

  a {
    color: ${theme.colors.grey50};
    font-weight: 600;

    &:hover {
      color: ${theme.colors.white100};
      transition: color .3s;
    }
  }
`;