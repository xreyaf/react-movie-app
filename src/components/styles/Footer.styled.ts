import styled from "@emotion/styled";
import { theme } from "../../styles/theme";

export const FooterStyled = styled.footer`
  background: ${theme.colors.grey900_80};
  backdrop-filter: blur(2px);
  padding: 5rem 0;
  text-align: center;
`;
