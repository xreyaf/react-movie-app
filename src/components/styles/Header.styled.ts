import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const HeaderStyled = styled.header`
  position: fixed;
width: 100%;
height: 10%;
  top: 0;
  left: 0;


  z-index: 100;
  background: ${theme.colors.grey900_80};
  backdrop-filter: blur(24px);

  

`;

export const Nav = styled.nav`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 3rem;

  ul {
    display: flex;
    flex-direction: row;
    justify-content: end;
    align-items: center;
  }
`;

export const Logo = styled(motion.img)`
  max-width: 3rem;
`;

export const StyledLinkMenu = styled(motion.a)`
  display: inline-flex;
  gap: 0.5rem;
  align-items: center;
  padding: 1rem 1rem;
  font-weight: 700;
  font-size: 1.2rem;
  cursor: pointer;
  color: ${theme.colors.grey400};

  &:hover {
    color: ${theme.colors.primary400};
  }
`;
