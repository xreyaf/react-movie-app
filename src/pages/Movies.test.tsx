import { theme } from '../styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { MemoryRouter } from 'react-router-dom';
import Movies from './Movies';
import { store } from '../app/store';
import { Provider } from 'react-redux';

test('it works', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <MemoryRouter>
          <Movies />
        </MemoryRouter>
      </Provider>
    </ThemeProvider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
