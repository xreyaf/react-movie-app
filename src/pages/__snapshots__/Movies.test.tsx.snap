// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.emotion-0 {
  width: 100%;
  max-width: 1440px;
  padding: 1rem 5rem;
  margin: 0 auto 2rem auto;
  height: 100%;
}

@media screen and (max-width: 1199.98px) {
  .emotion-0 {
    padding: 1rem 2rem;
  }
}

@media screen and (max-width: 991.98px) {
  .emotion-0 {
    padding: 1rem 1.5rem;
  }
}

@media screen and (max-width: 816.98px) {
  .emotion-0 {
    padding: 1rem 1.2rem;
  }
}

@media screen and (max-width: 575.98px) {
  .emotion-0 {
    padding: 1rem 1rem;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 3rem 0;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  >
    <div
      class="flapper__Wrapper-kzbmuk-0 jYJyDw"
      size="78"
    >
      <div
        class="flapper__Ball-kzbmuk-1 gTEOpN"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-26"
        y="13"
      />
      <div
        class="flapper__Ball-kzbmuk-1 cCTXgQ"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="4.491427816253161"
        y="-1.6838977275093896"
      />
      <div
        class="flapper__Ball-kzbmuk-1 lEfGR"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="12.022188575091121"
        y="-34.678316424296256"
      />
      <div
        class="flapper__Ball-kzbmuk-1 bgGXxN"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-9.078534174415228"
        y="-61.13778584819434"
      />
      <div
        class="flapper__Ball-kzbmuk-1 imBqUM"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-42.92146582558476"
        y="-61.13778584819435"
      />
      <div
        class="flapper__Ball-kzbmuk-1 eMxqxJ"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-64.02218857509112"
        y="-34.678316424296234"
      />
      <div
        class="flapper__Ball-kzbmuk-1 eyqkaq"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-56.491427816253164"
        y="-1.6838977275093967"
      />
    </div>
  </div>
</div>
`;
